tests/test_torch_core.py

```diff
@@ -19,3 +21,31 @@ def test_tensor_with_tensor():
     r = tensor(c)
     assert r.data_ptr()==c.data_ptr()
     assert torch.all(r==exp)
+
+def test_requires_grad():
+    m = simple_cnn(b)
+    assert requires_grad(m) == True
+
+def test_requires_grad_set():
+    m = simple_cnn(b)
+    requires_grad(m,False)
+    disj = False
+    for p in m.parameters():
```

Comment:
Those little for loops fit best in one single line. Can you edit your PR accordingly?